sw/device/mask_rom/mask_rom.c

```diff
@@ -3,13 +3,39 @@
 // SPDX-License-Identifier: Apache-2.0
 
 #include "sw/device/mask_rom/mask_rom.h"
+
+#include <stdbool.h>
+#include <stdint.h>
+
 #include "sw/device/lib/base/stdasm.h"
+#include "sw/device/lib/runtime/hart.h"
+#include "sw/device/rom_exts/rom_ext_manifest_parser.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+// This is the expected ROM_EXT Manifest Identifier
+static uint32_t const kRomExtIdentifierExpected = 0x4552544F;
+
+typedef void (*boot_fn)(void);
 
 void mask_rom_boot(void) {
-  asm volatile("unimp");
-  __builtin_unreachable();
+  rom_ext_manifest_t rom_ext = rom_ext_get_parameters(kRomExtManifestSlotA);
+
+  // Check we have a valid ROM_EXT
+  if (rom_ext_get_identifier(rom_ext) == kRomExtIdentifierExpected) {
+    // Set mtvec for ROM_EXT and jump to entry point.
+    asm volatile("csrw mtvec, %0" ::"r"(rom_ext.slot + 0x400u));
+
+    boot_fn rom_ext_entry = (boot_fn)(rom_ext.slot + 0x480u);
+    rom_ext_entry();
+  } else {
+    asm volatile("unimp");
```

Comment:
It's not `abort` in an effort to avoid dependencies this early. Eventually this will be a boot policy shutdown call, I expect (or a loop back around to the next slot).